src/verify_sig.rs

```diff
@@ -0,0 +1,58 @@
+use rsa::{RsaPrivateKey, RsaPublicKey};
+use rsa::pkcs1::{DecodeRsaPrivateKey, DecodeRsaPublicKey};
+use rsa::pkcs1v15::{SigningKey, VerifyingKey};
+use rsa::signature::{SignatureEncoding, Signer, Verifier};
+use rsa::sha2::Sha256;
+use std::{fs, str};
+
+// Takes a data buffer, a private key, and a public key, to sign the data
+// with the private key and verify the data with the public key.
+// Assume that we rely on RSA PKCS1v1.5, sha256 for the hash.
+pub fn verify_sig_rsa_pkcs1(databuf: &[u8], private_key: RsaPrivateKey, public_key: RsaPublicKey) {
```

Comment:
It would make more sense to have two functions:
```rust
pub fn verify_rsa_pkcs11(databuf: &[u8], signature: &[u8], public_key: RsaPublicKey) -> Result;
pub fn sign_rsa_pkcs11(databuf: &[u8], private_key: RsaPrivateKey) -> Result<Vec<u8>>;
```

Reply:
Done, thanks